exe-unit/components/gsb-http-proxy/src/monitor.rs

```diff
@@ -0,0 +1,11 @@
+pub trait RequestsMonitor: Sync + Send + Clone {
+    /// Called once on every HTTP request
+    #[allow(async_fn_in_trait)]
+    async fn on_request(&mut self) -> impl ResponseMonitor;
+}
+
+pub trait ResponseMonitor: Sync + Send {
+    /// Called once on HTTP response (for any response code) or on Drop.
+    #[allow(async_fn_in_trait)]
+    fn on_response(self);
+}
```

Comment:
As I commented in https://github.com/golemfactory/ya-runtime-ai/pull/75 implementations in this library should include more than just unimplemented abstractions

Reply:
This "more" could only mean an implementation of some "Counter" trait, which does not exist, and there is no obvious place to put it in. `ya-agreement-utils`? `ya-runtime-api`?
I just noticed that _Counters_ are  sometimes called _Metrics_. AI runtime initially had `duration_sec` implemented directly in main loop, without any dedicated struct or module, so I created one, but apparently it already existed as a [`TimeMETRIC`](https://github.com/golemfactory/yagna/blob/9fc18e4a10d4e4b69b8d4c4138a768cc0283b68d/exe-unit/src/metrics/mod.rs#L89).
If we want to have Counters as an API of a library then we should start by making public exe-unit's `Metric` trait together with some reusable implementations (like `duration_sec`).